test_conformance/subgroups/workgroup_common_templates.h

```diff
@@ -0,0 +1,585 @@
+//
+// Copyright (c) 2020 The Khronos Group Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//    http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+#ifndef WORKGROUPCOMMONTEMPLATES_H
+#define WORKGROUPCOMMONTEMPLATES_H
+
+#include "typeWrappers.h"
+#include <bitset>
+#include "CL/cl_half.h"
+
+
+typedef std::bitset<128> bs128;
+inline cl_uint4 generate_bit_mask(cl_uint subgroup_local_id,
+                                  std::string mask_type,
```

Comment:
Does this really need to be a copy? It seems to be that mask_type is not being modified.